File: lib/Models/OutletAddressDTO.cpp


#include "OutletAddressDTO.h"

using namespace Tiny;

OutletAddressDTO::OutletAddressDTO()
{
	regionId = long(0);
	street = std::string();
	number = std::string();
	building = std::string();
	estate = std::string();
	block = std::string();
	additional = std::string();
	km = int(0);
	city = std::string();
}

OutletAddressDTO::OutletAddressDTO(std::string jsonString)
{
	this->fromJson(jsonString);
}

OutletAddressDTO::~OutletAddressDTO()
{

}

void
OutletAddressDTO::fromJson(std::string jsonObj)
{
    bourne::json object = bourne::json::parse(jsonObj);

    const char *regionIdKey = "regionId";

    if(object.has_key(regionIdKey))
    {
        bourne::json value = object[regionIdKey];



        jsonToValue(&regionId, value, "long");


    }

    const char *streetKey = "street";

    if(object.has_key(streetKey))
    {
        bourne::json value = object[streetKey];



        jsonToValue(&street, value, "std::string");


    }

    const char *numberKey = "number";

    if(object.has_key(numberKey))
    {
        bourne::json value = object[numberKey];



        jsonToValue(&number, value, "std::string");


    }

    const char *buildingKey = "building";

    if(object.has_key(buildingKey))
    {
        bourne::json value = object[buildingKey];



        jsonToValue(&building, value, "std::string");


    }

    const char *estateKey = "estate";

    if(object.has_key(estateKey))
    {
        bourne::json value = object[estateKey];



        jsonToValue(&estate, value, "std::string");


    }

    const char *blockKey = "block";

    if(object.has_key(blockKey))
    {
        bourne::json value = object[blockKey];



        jsonToValue(&block, value, "std::string");


    }

    const char *additionalKey = "additional";

    if(object.has_key(additionalKey))
    {
        bourne::json value = object[additionalKey];



        jsonToValue(&additional, value, "std::string");


    }

    const char *kmKey = "km";

    if(object.has_key(kmKey))
    {
        bourne::json value = object[kmKey];



        jsonToValue(&km, value, "int");


    }

    const char *cityKey = "city";

    if(object.has_key(cityKey))
    {
        bourne::json value = object[cityKey];



        jsonToValue(&city, value, "std::string");


    }


}

bourne::json
OutletAddressDTO::toJson()
{
    bourne::json object = bourne::json::object();





    object["regionId"] = getRegionId();






    object["street"] = getStreet();






    object["number"] = getNumber();






    object["building"] = getBuilding();






    object["estate"] = getEstate();






    object["block"] = getBlock();






    object["additional"] = getAdditional();






    object["km"] = getKm();






    object["city"] = getCity();



    return object;

}

long
OutletAddressDTO::getRegionId()
{
	return regionId;
}

void
OutletAddressDTO::setRegionId(long  regionId)
{
	this->regionId = regionId;
}

std::string
OutletAddressDTO::getStreet()
{
	return street;
}

void
OutletAddressDTO::setStreet(std::string  street)
{
	this->street = street;
}

std::string
OutletAddressDTO::getNumber()
{
	return number;
}

void
OutletAddressDTO::setNumber(std::string  number)
{
	this->number = number;
}

std::string
OutletAddressDTO::getBuilding()
{
	return building;
}

void
OutletAddressDTO::setBuilding(std::string  building)
{
	this->building = building;
}

std::string
OutletAddressDTO::getEstate()
{
	return estate;
}

void
OutletAddressDTO::setEstate(std::string  estate)
{
	this->estate = estate;
}

std::string
OutletAddressDTO::getBlock()
{
	return block;
}

void
OutletAddressDTO::setBlock(std::string  block)
{
	this->block = block;
}

std::string
OutletAddressDTO::getAdditional()
{
	return additional;
}

void
OutletAddressDTO::setAdditional(std::string  additional)
{
	this->additional = additional;
}

int
OutletAddressDTO::getKm()
{
	return km;
}

void
OutletAddressDTO::setKm(int  km)
{
	this->km = km;
}

std::string
OutletAddressDTO::getCity()
{
	return city;
}

void
OutletAddressDTO::setCity(std::string  city)
{
	this->city = city;
}

File: lib/app/Models/OutletAddressDTO.php
<?php
/**
 * OutletAddressDTO
 */
namespace app\Models;

/**
 * OutletAddressDTO
 * @description Адрес точки продаж.
 */
class OutletAddressDTO {

    /** @var int $regionId Идентификатор региона.  Идентификатор можно получить c помощью запроса [GET regions](../../reference/regions/searchRegionsByName.md).  {% note alert %}  При создании и редактировании точек продаж можно указывать только регионы типов &#x60;TOWN&#x60; (город), &#x60;CITY&#x60; (крупный город) и &#x60;REPUBLIC_AREA&#x60; (район субъекта федерации). Тип региона указан в выходных параметрах type запросов [GET regions](../../reference/regions/searchRegionsByName.md) и [GET regions/{regionId}](../../reference/regions/searchRegionsById.md).  {% endnote %}*/
    public $regionId = 0;

    /** @var string $street Улица.*/
    public $street = "";

    /** @var string $number Номер дома.*/
    public $number = "";

    /** @var string $building Номер строения.*/
    public $building = "";

    /** @var string $estate Номер владения.*/
    public $estate = "";

    /** @var string $block Номер корпуса.*/
    public $block = "";

    /** @var string $additional Дополнительная информация.*/
    public $additional = "";

    /** @var int $km Порядковый номер километра дороги, на котором располагается точка продаж, если отсутствует улица.*/
    public $km = 0;

    /** @var string $city {% note warning \&quot;\&quot; %}  Этот параметр устарел. Не используйте его. Город или населенный пункт возвращается в параметре &#x60;regionId&#x60;.  {% endnote %}*/
    public $city = "";

}
